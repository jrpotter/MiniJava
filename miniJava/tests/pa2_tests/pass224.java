//PA2 pass qualified reference expression
class A {
    void p(){
        int a = x[3].y[4](z,5) + 2;
    }
}
