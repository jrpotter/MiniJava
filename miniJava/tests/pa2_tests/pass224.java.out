======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     INT BaseType
  .   .   .     "a" varname
  .   .   .   BinaryExpr
  .   .   .     "+" Operator
  .   .   .       CallExpr
  .   .   .         IndexedRef
  .   .   .           LiteralExpr
  .   .   .             "4" IntLiteral
  .   .   .           QualifiedRef
  .   .   .             "y" Identifier
  .   .   .             IndexedRef
  .   .   .               LiteralExpr
  .   .   .                 "3" IntLiteral
  .   .   .               IdRef
  .   .   .                 "x" Identifier
  .   .   .         ExprList + [2]
  .   .   .         . RefExpr
  .   .   .         .   IdRef
  .   .   .         .     "z" Identifier
  .   .   .         . LiteralExpr
  .   .   .         .   "5" IntLiteral
  .   .   .       LiteralExpr
  .   .   .         "2" IntLiteral
=============================================
